### PR TITLE
feat: Implemented Scrolling to move between pages

### PR DESCRIPTION
**Pull request by M4dhav**
Implemented scrolling as required in #4

### CODE REVIEW

**Comment by guyluz11** [comment]
Have you tested extending scroll physics as I suggested in the issue?
It could be much cleaner and understandable code.

**Comment by M4dhav** [comment]
Where do you want to pass ScrollPhysics? As far as I can see, IntroductionScreen doesn't have a named parameted scrollPhysics

**Comment by guyluz11** [comment]
It exists and already set to not scroll at all 
https://github.com/guyluz11/infinite_horizons/blob/45880dafe208ae303ee9b2e7c311999c5720ea7e/lib/presentation/pages/intro_page.dart#L99

**Comment by M4dhav** [comment]
Oh! I completely missed that. I'll reimplement this with Scroll Physics

**Comment by guyluz11** [comment]
Make sure to also lock the scroll option in the last intro page.
After he clicks the "I am ready" button and there is confetti animation and then he is being redirected to the timer page.

**Review by guyluz11 — CHANGES_REQUESTED** [review]
Is there any news about the needed progress here?

**Comment by M4dhav** [comment]
I will get back to it within the week, I already did try with Scroll Physics once before but it had some issues (pages getting stuck between scrolls). I will try and sort them out soon

**Comment by M4dhav** [comment]
@guyluz11 I've looked into this again and due to lack of documentation and code examples for the implementation of ScrollPhysics that we need, I am unable to sort out the issue that is present during the use of scroll physics. It works but it looks weird and bad. I think the GestureDetector implementation is the best option for us. 
Please let me know how you want to proceed.

**Comment by guyluz11** [comment]
ScrollPhysics code shold be cleener and better option, so let me try to implement it as well.
In the mentime feel free work on other issues.

**Comment by guyluz11** [comment]
I tried it and managed to make it work, but it looks like the implementation is not as clean as I thought it would be.

So let's proceed with GestureDetector

**Comment by M4dhav** [comment]
Okay, I will resolve conflicts and get it ready for a merge

**Comment by guyluz11** [comment]
> Okay, I will resolve conflicts and get it ready for a merge

Good timing :), I have just pushed another pr and wanting to upload a new version.
Should I wait for this pr?, what is your time estimation.

**Comment by M4dhav** [comment]
Finally finished, sorry it took this long, College started back up and I have been super busy

File: lib/presentation/pages/intro_page.dart [hunk 1]
```diff
@@ -52,6 +52,20 @@ class _IntroPageState extends State<IntroPage> {
...
+      return; // user have just tapped on screen (no dragging)
+    }
+
+    if (details.primaryVelocity!.compareTo(0) == -1) {
+      if (showNextButton) {
+        nextPage();
+      }
+    } else {
+      previousPage();
+    }
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:66]: Please remove the else and only call previousPage

File: lib/presentation/pages/intro_page.dart [hunk 2]
```diff
@@ -52,6 +52,20 @@ class _IntroPageState extends State<IntroPage> {
...
 
+  void onHorizontalDrag(DragEndDetails details) {
+    if (details.primaryVelocity == 0) {
+      return; // user have just tapped on screen (no dragging)
+    }
+
+    if (details.primaryVelocity!.compareTo(0) == -1) {
+      if (showNextButton) {
+        nextPage();
+      }
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:63]: Add return

File: lib/presentation/pages/intro_page.dart [hunk 3]
```diff
@@ -52,6 +52,20 @@ class _IntroPageState extends State<IntroPage> {
 
   void previousPage() => _introKey.currentState?.previous();
 
+  void onHorizontalDrag(DragEndDetails details) {
+    if (details.primaryVelocity == 0) {
+      return; // user have just tapped on screen (no dragging)
+    }
+
+    if (details.primaryVelocity!.compareTo(0) == -1) {
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:60]: Change to else if

File: lib/presentation/pages/intro_page.dart [hunk 4]
```diff
@@ -76,63 +90,67 @@ class _IntroPageState extends State<IntroPage> {
...
-          ),
-          overrideNext: Center(
-            child: ButtonAtom(
-              variant: ButtonVariant.highEmphasisFilled,
-              onPressed: showNextButton ? nextPage : () {},
-              icon: FontAwesomeIcons.arrowRight,
-              text: 'next',
-              disabled: !showNextButton,
+        child: GestureDetector(
+          onHorizontalDragEnd: (DragEndDetails details) =>
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:94]: Please change to 
onHorizontalDragEnd: onHorizontalDrag

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by guyluz11** [comment]
> Finally finished, sorry it took this long, College started back up and I have been super busy

Good to have you back 😌

I hope you are doing great at college

**Comment by M4dhav** [comment]
> I hope you are doing great at college

5th Semester is quite a bit harder than everything before but I'm handling it pretty well

**Review by guyluz11 — APPROVED** [review]

**Pull request merged**